src/inmanta/util.py

```diff
@@ -487,17 +493,32 @@ async def retry_limited(
     *args: object,
     **kwargs: object,
 ) -> None:
+    """
+    This function makes use of the INMANTA_RETRY_LIMITED_MULTIPLIER env variable. If set, INMANTA_RETRY_LIMITED_MULTIPLIER
+    serves as multiplier: The timeout given as argument becomes a 'soft limit' and the "soft limit' mutliplied by the
+    multiplier (form the env var) becomes a "hard limit". If the hard limit is reached before the wait condition is fulfilled
+    a Timeout exception is raised. If the wait condition is fulfilled before the hard limit is reached but after the soft
+    limit is reached, a different Timeout exception is raised. if the Env var is not set, then the soft and hard limit are
+    the same.
+    """
+
     async def fun_wrapper() -> bool:
         if inspect.iscoroutinefunction(fun):
             return await fun(*args, **kwargs)
         else:
             return fun(*args, **kwargs)
 
+    multiplier: int = int(os.environ.get("INMANTA_RETRY_LIMITED_MULTIPLIER", 1))
+    hard_timeout = timeout * multiplier
     start = time.time()
-    while time.time() - start < timeout and not (await fun_wrapper()):
+    while time.time() - start < hard_timeout and not (await fun_wrapper()):
         await asyncio.sleep(interval)
     if not (await fun_wrapper()):
-        raise asyncio.TimeoutError()
+        raise Timeout(f"Wait condition was not reach after hard limit of {hard_timeout} seconds")
+    if time.time() - start > timeout:
+        raise Timeout(
+            f"Wait condition was met after {time.time() - start} seconds, but soft limit was set to {timeout} seconds"
+        )
```

Comment:
This method is used pretty much everywhere and some test cases might rely on the fact that an `asyncio.TimeoutError` is raised. Changing the type of the exception might break tests all over the place.

Reply:
You mean outside of core? I'll set the old exception back

Reply:
Yes. This method is packaged into `pytest-inmanta-extensions`. As such, it can be used by the test suite of any extension or module.